pkg/controller/trainingjob_controller.go

```diff
@@ -0,0 +1,258 @@
+package controller
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	extv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	extclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	paddlev1 "github.com/paddlepaddle/edl/pkg/apis/paddlepaddle/v1"
+	paddleclient "github.com/paddlepaddle/edl/pkg/client/clientset/versioned"
+	paddlescheme "github.com/paddlepaddle/edl/pkg/client/clientset/versioned/scheme"
+	paddleinformers "github.com/paddlepaddle/edl/pkg/client/informers/externalversions"
+	paddlelisters "github.com/paddlepaddle/edl/pkg/client/listers/paddlepaddle/v1"
+	"github.com/paddlepaddle/edl/pkg/updater"
+)
+
+type TrainingJobController struct {
```

Comment:
Exported type must have comments. Curious this should be checked by the linter.

Reply:
Right, I will supplement some comments.